generator/internal/dart/dart.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dart
+
+import (
+	"embed"
+	"fmt"
+	"log/slog"
+	"strings"
+	"unicode"
+
+	"github.com/googleapis/google-cloud-rust/generator/internal/api"
+	"github.com/googleapis/google-cloud-rust/generator/internal/language"
+	"github.com/iancoleman/strcase"
+)
+
+//go:embed templates
+var dartTemplates embed.FS
+
+type goImport struct {
+	path string
+	name string
+}
+
+func Generate(model *api.API, outdir string, options map[string]string) error {
+	_, err := annotateModel(model, options)
+	if err != nil {
+		return err
+	}
+	provider := templatesProvider()
+	return language.GenerateFromRoot(outdir, model, provider, generatedFiles())
+}
+
+func generatedFiles() []language.GeneratedFile {
+	return language.WalkTemplatesDir(dartTemplates, "templates")
+}
+
+func loadWellKnownTypes(s *api.APIState) {
+	timestamp := &api.Message{
+		ID:      ".google.protobuf.Timestamp",
+		Name:    "Time",
+		Package: "time",
+	}
+	duration := &api.Message{
+		ID:      ".google.protobuf.Duration",
+		Name:    "Duration",
+		Package: "time",
+	}
+	s.MessageByID[timestamp.ID] = timestamp
+	s.MessageByID[duration.ID] = duration
+}
+
+func fieldType(f *api.Field, state *api.APIState, importMap map[string]*goImport) string {
+	var out string
+	switch f.Typez {
+	case api.STRING_TYPE:
+		out = "string"
+	case api.INT64_TYPE:
+		out = "int64"
+	case api.INT32_TYPE:
+		out = "int32"
+	case api.BOOL_TYPE:
+		out = "bool"
+	case api.BYTES_TYPE:
+		out = "[]byte"
+	case api.MESSAGE_TYPE:
+		m, ok := state.MessageByID[f.TypezID]
+		if !ok {
+			slog.Error("unable to lookup type", "id", f.TypezID)
+			return ""
+		}
+		if m.IsMap {
+			key := fieldType(m.Fields[0], state, importMap)
+			val := fieldType(m.Fields[1], state, importMap)
+			out = "map[" + key + "]" + val
```

Comment:
This is very go-like code.  I think we can remove it and add some bug to track it?